src/build/app/debug_asserts.rs

```diff
@@ -228,6 +228,25 @@ pub(crate) fn assert_app(app: &App) {
                 arg
             )
         }
+
+        //required groups shouldn't have args with default values
+        if group.required {
+            for arg in &group.args {
```

Comment:
You don't need this line if you move the `if group.required` into the for loop described above